src/AlgebraicGeometry/Schemes/AffineSchemes/Objects/Attributes.jl

```diff
@@ -493,14 +493,18 @@ with coordinates
   I = modulus(OO(X))
   J = radical(pre_saturated_ideal(I))
   inc = ClosedEmbedding(X, ideal(OO(X), OO(X).(gens(J))))
-  return domain(inc), inc
+  Xred, inc = domain(inc), inc
+  set_attribute!(Xred, :is_reduced=>true)
+  return Xred, inc
   return Spec(base_ring(J), J, inverted_set(OO(X)))
 end
```

Comment:
Please check here, whether is_reduced is set for X and bypass the radical in this case.
Do we need a ``check`` here?

Reply:
where and for what?

Reply:
We are entering reduced_scheme with a scheme X. 
If is_reduced happens to be set for X, we can avoid the whole computation of the radical. In this case, ``check`` could control whether we trust this and simply set ``Xred, inc = X, identity`` or whether we compute the radical anyway. 
(I am not talking about @check but about check)

Reply:
Such an argument would be called `recompute`. But it would result in some trouble w.r.t. caching.